### PR TITLE
Changed Pipeline creation to fail-fast fashion

### PR DESCRIPTION
**Pull request by templatevk**

### CODE REVIEW

File: sawmill-core/src/test/java/io/logz/sawmill/PipelineTest.java [hunk 1]
```diff
@@ -185,21 +184,11 @@ public void shouldFailPipelineCreationOnNoProcessorFactoryConfigurationGiven() t
...
-    @Test(expected = FactoryInstantiationException.class)
     public void shouldFailPipelineCreationOnGeoIpConfigurationInvalidFileGiven() throws Exception {
         new Pipeline.Factory(new GeoIpConfiguration("LICENSE"));
     }
 
-    @Test(expected = FactoryInstantiationException.class)
+    @Test(expected = SawmillException.class)
     public void shouldFailPipelineCreationOnGeoIpConfigurationFileDoesNotExist() throws Exception {
         new Pipeline.Factory(new GeoIpConfiguration("non-existing-file"));
     }
```
Comment (naorgu) [inline sawmill-core/src/test/java/io/logz/sawmill/PipelineTest.java:194]: `TestProcessorWithMandatoryDependencies` is unused, let's delete

File: sawmill-core/src/main/java/io/logz/sawmill/ProcessorFactoriesLoader.java [hunk 2]
```diff
@@ -43,16 +40,14 @@ public void loadAnnotatedProcessors(ProcessorFactoryRegistry processorFactoryReg
...
                 ProcessorProvider processorProvider = processor.getAnnotation(ProcessorProvider.class);
                 String typeName = processorProvider.type();
-                processorFactoryRegistry.register(typeName, getFactory(processorProvider));
-                logger.debug("{} processor factory loaded successfully, took {}ms", typeName, stopwatch.elapsed(MILLISECONDS) - timeElapsed);
-                processorsLoaded++;
-            } catch(InvocationTargetException e) {
-                if (e.getCause() instanceof FactoryInstantiationException) {
-                    throw new FactoryInstantiationException(
-                            String.format("Could not instantiate %s processor factory", processor.getName()), e);
+                Processor.Factory factory = getFactory(processorProvider);
```
Comment (naorgu) [inline sawmill-core/src/main/java/io/logz/sawmill/ProcessorFactoriesLoader.java:43]: Let's use optional
```suggestion
                Optional<Processor.Factory> factory = getFactory(processorProvider);
```

Comment (naorgu) [inline sawmill-core/src/main/java/io/logz/sawmill/ProcessorFactoriesLoader.java:43]: `getFactory` method can be private

File: sawmill-core/src/test/java/io/logz/sawmill/PipelineTest.java (hunk 1 above)
Comment (templatevk) [reply sawmill-core/src/test/java/io/logz/sawmill/PipelineTest.java:194]: its deleted already

**Pull request merged**